spec/models/user_spec.rb

```diff
@@ -1,41 +1,55 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
   describe 'Factory' do
+    subject { build(:user) }
+
     it 'has a valid factory' do
-      # Testear que el factory definido es válido.
+      is_expected.to be_valid
     end
   end
 
   describe 'Associations' do
-    # Testear asociaciones (shoulda-matchers).
-    # https://github.com/thoughtbot/shoulda-matchers#activerecord-matchers
+    it { should have_many(:posts).dependent(:destroy) }
   end
 
   describe 'Presence validations' do
-    # Testear validaciones de presencia (shoulda-matchers).
-    # https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers
+    it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:password) }
+    it { should validate_presence_of(:role) }
   end
 
   describe 'Uniqueness validations' do
-    # Testear validaciones de unicidad (shoulda-matchers).
-    # https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers
+    subject { create(:user) }
+
+    it { should validate_uniqueness_of(:email) }
   end
 
   describe 'Length validations' do
-    # Testear validaciones de longitud (shoulda-matchers).
-    # https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers
+    it { should validate_length_of(:password).is_at_least(8) }
   end
 
   describe 'Enumeratives' do
-    # Testear definición de enumerativos (shoulda-matchers).
+    it { should define_enum_for(:role).with_values([:admin, :regular]) }
   end
 
-  # Testear métodos de instancia y de clase como para el caso de cualquier
-  # otra clase Ruby (similar a testear ApiLoginManager).
   describe '#valid_password?' do
-    # Testear funcionamiento de método. Podemos definir dos contexts:
-    #  - 'when given value is different from password'
-    #  - 'when given value is equal to password'
+    subject { user.valid_password?(password) }
+
+    let(:user) { create(:user, password: 'password_secret') }
+
+    context 'when given value is different from password' do
+      let(:password) { 'password_wrong' }
+
+      it { is_expected.to be_falsey }
```

Comment:
Está bien, pero sabemos que si falla el resultado es `false`, por lo que creo que sería más específico con `be false`.